public/app/features/live/live.ts

```diff
@@ -114,9 +115,19 @@ class CentrifugeSrv implements GrafanaLiveSrv {
   }
 
   initChannel<T>(path: string, handler: ChannelHandler<T>) {
+    const connected = from(
+      new Promise<void>(resolve => {
+        if (this.centrifuge.isConnected()) {
+          return resolve();
+        }
+
+        this.centrifuge.addListener('connect', resolve);
```

Comment:
you should be able to define this promise in the constructor and reuse it

Reply:
as a promise that has completed can be reused  (and will complete with same value)